brave/brave-ios
# Fix #1514: Add ability to automatically close stale tabs.

## Summary of Changes


This pull request fixes #1514 

## Submitter Checklist:


## Test Plan:


One note for existing tabs before you update to this build:
Such tabs will be never cleared until the user has visited them at least once

Fresh install test
- Fresh install 1.26
- Go to settings->Close tabs, choose ‚after one day’
- Open few tabs with random urls
- Leave the app for 5+ minutes
- Kill the app and relaunch
- Verify that tabs got removed

Fresh install, one tab active
- Fresh install 1.26
- Go to settings->Close tabs, choose ‚after one day’
- Open few tabs with random urls
- Wait 5 minutes, don’t close the app
- Go to one tab to make it ‚active’, remember this tab
- Kill the app, relaunch
- Verify that the tab that you selected restored correctly

Upgrade 
- Install 1.25
- Open few tabs with random urls
- Leave the app for 5+ minutes
- Upgrade to 1.26
- Verify that all tabs are restored

For upgrades there is one edge case, old tabs pre 1.26 are never purged if you don’t visit them. I do not think this needs testing

## Screenshots:



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
